src/control/IntersectionReferenceSystem.ts

```diff
@@ -196,6 +200,49 @@ export class IntersectionReferenceSystem {
     return { points, offset };
   }
 
+  /**
+   * Generate a set of coordinates along the trajectory of the curve
+   */
+  getExtendedTrajectory(steps: number, extensionStart = DEFAULT_START_EXTEND_LENGTH, extensionEnd = DEFAULT_END_EXTEND_LENGTH): Trajectory {
+    if (!extensionStart || extensionStart < 0.0) {
+      throw new Error('Invalid parameter, getExtendedTrajectory() must be called with a positive extensionStart parameter');
+    }
+    if (!extensionEnd || extensionEnd < 0.0) {
+      throw new Error('Invalid parameter, getExtendedTrajectory() must be called with a positive extensionEnd parameter');
+    }
+
+    const totalLength = this.displacement + extensionStart + extensionEnd;
+    const preSteps = Math.floor((extensionStart / totalLength) * steps);
+    const curveSteps = Math.ceil((this.displacement / totalLength) * steps);
+    const postSteps = steps - curveSteps - preSteps;
+
+    const points = [];
+
+    const refStart = new Vector2(this.interpolators.trajectory.getPointAt(0.0));
+    const startVec = new Vector2(this.startVector);
```

Comment:
should the static `getDirectionVector() return a Vector2 object instead of an array of number (number[]), and by association the startVector and endVector?

Reply:
I think it's more flexible if input/output to/from the component is simple structures so we don't force users to use Vector2.

Reply:
that's fair